src/lib/methods/batch/batch-write.ts

```diff
@@ -0,0 +1,85 @@
+import { Executable } from '../executable';
+import { DynamoDB } from '../../dynamodb';
+import { BatchItem } from './batch-item';
+import { BatchMethod } from './batch-method';
+import { BatchWriteItemInput } from 'aws-sdk/clients/dynamodb';
+import pRetry from 'p-retry';
+
+export class BatchWrite extends BatchMethod implements Executable {
+
+	constructor(
+		dynamodb: DynamoDB,
+		private items: BatchItem[],
+		private retryCount = 5,
+		private randomize = true) {
```

Comment:
Perhaps we should call this `randomizeRetry` or something if we really want to provide this to the constructor. It isn't really clear what this `randomize` actually randomizes imho